lib/mobility-core/src/Kernel/External/Maps/Interface.hs

```diff
@@ -55,11 +55,14 @@ getDistance ::
   ) =>
   MapsServiceConfig ->
   GetDistanceReq a b ->
+  (NonEmpty (Meters, Seconds) -> (Meters, Seconds)) ->
   m (GetDistanceResp a b)
-getDistance serviceConfig GetDistanceReq {..} =
-  getDistances serviceConfig getDistancesReq >>= \case
-    (a :| []) -> return a
-    _ -> throwError (InternalError "Exactly one getDistance result expected.")
+getDistance serviceConfig req@GetDistanceReq {..} dataDecider = case serviceConfig of
+  GoogleConfig cfg -> Google.getDistance cfg req dataDecider
+  _ ->
```

Comment:
Let's not fallback to distance matrix API here. If config for directions API is provided for other vendors which is not implemented then we should fail